src/lib/bootloader/mbr_update.rb

```diff
@@ -183,8 +156,8 @@ def activatable_partitions(disk)
       disk.partitions.reject { |p| p.id.is?(:swap, :bios_boot) }
     end
 
-    def extended_partition(disk)
-      part = activatable_partitions(disk).find { |p| p.type == Y2Storage::PartitionType::EXTENDED }
+    def extended_partition(partition)
+      part = partition.siblings.find { |p| p.type.is?(:extended) }
```

Comment:
Of course, `partition.disk` returns the disk for any partition. I don't see why the fact that the disk is not it's direct parent should change that. In fact, the disk is not the direct parent for primary partitions either. The direct parent would be a `PartitionTable` object in that case.